### DISCUSSION TITLE
Autoselect Template

### DISCUSSION THREAD

**Discussion by narendra-bhole** [chezou/tabula-py#367] [Ideas]
I have multiple templates generated from different Invoice Patterns. These templates generated from tabula ui. Now In python program with read_pdf_with_template function, I would like to automatically select template from available templates folder. Please let me know if any one has idea how to do it.

**Comment by chezou** [comment]
Hi @narendra-bhole, thanks for sharing your idea.

I don't understand your request properly yet, but can you share a specific example? What kind of "automatically" do you expect? What kind of data can be used for selecting a template?

**Reply by narendra-bhole** [reply]
I possess numerous invoices, each adhering to a specific format that aligns with a corresponding template file. Upon uploading an invoice for OCR processing, the system should automatically select the appropriate template and execute OCR to extract text from designated areas.

**Reply by chezou** [reply]
Thanks for your explanation. 

However, I still don't understand how to select an appropriate template, even by a human. It sounds like tabula-py will **automagically** select the template without knowing any prior information, which is hard to solve with my knowledge since there's no metadata for the template, nor PDF.

**Reply by narendra-bhole** [reply]
Currently I'm doing with keyword => Template file data. Searching Keyword in PDF and select template based on keyword. I would like to further extend it without keyword search. Can be using ML models. Template metadata can be set of areas selelcted in Template file. e.g. Template 1 has 10 areas and Template 2 has 15 different areas. So with this we can train and create models.

**Reply by chezou** [reply]
Thanks for sharing the details.

As an ML engineer, I can say we need to have hundreds or a thousand labeled data sets with carefully considered features as input data with more solid requirements.

You can try to build a heuristic rule based on your idea, and then you can build your own ML models. And that is, unfortunately, out of the scope of tabula-py.
